### PR TITLE
better learners all tests (MERGE SOON)

### PR DESCRIPTION
**Pull request by berndbischl**
obvious test code improvement

might be buggy, but travis will tell

pls review then merge when ok and green

### CODE REVIEW

**Comment by jakob-r** [comment]
Can't see anything bad here. Might have used `checkmate::test_*` in some cases.

**Comment by berndbischl** [comment]
> Can't see anything bad here. Might have used checkmate::test_* in some cases.

well, travis failed. and pls give conrete hints what function you mean from cm.

File: tests/testthat/helper_learners_all.R [hunk 1]
```diff
@@ -55,23 +55,34 @@ testThatLearnerRespectsWeights = function(lrn, task, train.inds, test.inds, weig
...
-  
-  if (pred.type == "se")
-    expect_equal(length(p$data$se), getTaskSize(task))
-  
+
+  if (pred.type == "se") {
+    s = p$data$se
+    expect_is(s, "numeric")
+    expect_length(s, getTaskSize(task))
+    expect_true(all(s >= 0))
```
Comment (jakob-r) [inline tests/testthat/helper_learners_all.R]: Last three lines
```r
expect_numeric(s, lower = 0, finite = TRUE, any.missing = FALSE, len = getTaskSize(task))
```

File: tests/testthat/helper_learners_all.R [hunk 2]
```diff
@@ -55,23 +55,34 @@ testThatLearnerRespectsWeights = function(lrn, task, train.inds, test.inds, weig
...
   if (pred.type == "prob") {
-    expect_false(anyNA(getPredictionProbabilities(p)))
-    expect_equal(NROW(getPredictionProbabilities(p)), getTaskSize(task))
+    cls = getTaskClassLevels(task)
+    pmat = getPredictionProbabilities(p, cl = cls)
+    expect_is(pmat, "data.frame")
+    expect_equal(NROW(pmat), getTaskSize(task))
+    expect_equal(NCOL(pmat), length(cls))
+    expect_true(setequal(colnames(pmat), cls))
+    expect_false(anyNA(pmat))
```
Comment (jakob-r) [inline tests/testthat/helper_learners_all.R]: Last 5 lines:
```r
expect_data_frame(pmat, types = "numeric", any.missing = FALSE, nrows = getTaskSize(task), ncols = length(cls), col.names = cls)
```

**Comment by berndbischl** [comment]
@jakob-r 
thx a lot. this is indeed much better. i need more practice with the test functions from cm
will try to read more about them and change this

**Comment by jakob-r** [comment]
Added line notes because I didn't want to interfere with your edits. checkmate works best with autocompletion ;)

**Comment by berndbischl** [comment]
> checkmate works best with autocompletion - zero learning needed ;)

i need to at least learn that certain things can be done easier and in "bulk"....

**Comment by berndbischl** [comment]
@jakob-r @mllg 

expect_data_frame(pmat, types = "numeric", any.missing = FALSE, nrows = getTaskSize(task), ncols length(cls), col.names = cls)

note that the colnames stuff does not work that you posted. so there is even more learning required. 
i suggested to michel a couple of times that it should be possible to pass a vector of colnames here.
but you can only set stuff like "named". you need to do a 2nd call with checkNamed.

**Pull request merged**